pyomo/util/check_units.py

```diff
@@ -240,11 +241,7 @@ def assert_units_consistent(obj):
     if objtype in native_types:
         return
     elif obj.is_expression_type() or objtype is IndexTemplate:
-        try:
-            _assert_units_consistent_expression(obj)
-        except UnitsError:
-            print('Units problem with expression {}'.format(obj))
```

Comment:
I agree with removing these print statements, but should the message still be sent out the logger?  (i.e., I don't know if the exception generated within the walker contains enough information - for example, does it include the constraint / objective name?